### PR TITLE
Upgrade PHPCS standards to WPCS 2.1

### PR DESCRIPTION
**Pull request by jom**
This upgrades the PHPCS standards to WPCS 2.1 and includes some less controversial fixes (a lot of whitespace changes and `isset()` checks). 
 
This involves a number of composer updates, so make sure you run `composer install` on this PR.

To keep this PR from getting massive, I've hidden some of the rules under `severity=4`. You can see these errors and warnings if you run:
` phpcs -p --severity=4 -s phpcs.xml.dist .`

After this PR is merged (post-1.33.0 release), I'll create issues to remove the remaining rules.

### Post Merge

Create issues to fix rule breaks and remove exclusions for:
- [ ] `WordPress.WP.GlobalVariablesOverride.Prohibited`: Most of these seem to be naming conflicts for variables. Some are indeed global overrides.
- [ ] `WordPress.Security.ValidatedSanitizedInput.MissingUnslash`: Lots of changes, but pretty uncontroversial. 
- [ ] `WordPress.Security.NonceVerification`: I foresee a lot of `phpcs:ignore` comments :)
- [ ] `WordPress.Security.ValidatedSanitizedInput.InputNotSanitized`: I foresee a lot of `phpcs:ignore` comments :)
- [ ] `WordPress.DB.DirectDatabaseQuery`: Probably a lot of `phpcs:ignore`.
- [ ] `VariableAnalysis`: Some valid fixes required here. Some of these variables go into templates.

### CODE REVIEW

File: phpcs.xml.dist [hunk 1]
```diff
@@ -1,62 +1,81 @@
 <?xml version="1.0"?>
 <ruleset name="WP Job Manager">
-	<!-- See https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml -->
-	<!-- See https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress-Core/ruleset.xml -->
-
-	<!-- Set a description for this ruleset. -->
 	<description>A custom set of code standard rules to check for WordPress themes and plugins.</description>
 
-	<config name="installed_paths" value="vendor/wp-coding-standards/wpcs,vendor/phpcompatibility/php-compatibility" />
+	<config name="installed_paths" value="vendor/wp-coding-standards/wpcs,vendor/phpcompatibility/phpcompatibility-paragonie,vendor/phpcompatibility/php-compatibility,vendor/phpcompatibility/phpcompatibility-wp,vendor/sirbrillig/phpcs-variable-analysis" />
```
Comment (GaryJones) [inline phpcs.xml.dist]: If you add https://packagist.org/packages/dealerdirect/phpcodesniffer-composer-installer to your require-dev, then standards will get automatically registered on `composer update` and `composer install`. No need for this `installed_paths` config.

File: phpcs.xml.dist [hunk 2]
```diff
@@ -1,62 +1,81 @@
...
 	<!-- Configs -->
 	<config name="minimum_supported_wp_version" value="4.7" />
 	<config name="testVersion" value="5.2-"/>
 
 	<!-- Rules -->
-	<rule ref="WordPress.Files.FileName.InvalidClassFileName">
-		<exclude-pattern>includes/**/abstract-*.php</exclude-pattern>
-		<exclude-pattern>tests/*</exclude-pattern>
+	<rule ref="PHPCompatibilityWP"/>
+	<rule ref="WordPress-Core" />
```
Comment (GaryJones) [inline phpcs.xml.dist]: No need to include `WordPress-Core`, when you have `Wordpress-Extra`, as Extra already includes Core.

File: phpcs.xml.dist [hunk 3]
```diff
@@ -1,62 +1,81 @@
...
-	<!-- See https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml -->
-	<!-- See https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress-Core/ruleset.xml -->
-
-	<!-- Set a description for this ruleset. -->
 	<description>A custom set of code standard rules to check for WordPress themes and plugins.</description>
 
-	<config name="installed_paths" value="vendor/wp-coding-standards/wpcs,vendor/phpcompatibility/php-compatibility" />
+	<config name="installed_paths" value="vendor/wp-coding-standards/wpcs,vendor/phpcompatibility/phpcompatibility-paragonie,vendor/phpcompatibility/php-compatibility,vendor/phpcompatibility/phpcompatibility-wp,vendor/sirbrillig/phpcs-variable-analysis" />
+
+	<arg value="s"/>
```
Comment (GaryJones) [inline phpcs.xml.dist]: Consider adding `p` to this value as well - it provides a progress indicator.

File: phpcs.xml.dist (hunk 1 above)
Comment (jom) [reply phpcs.xml.dist]: We already use that package. I'm not sure why we maintain it here as well. I can remove it.

**Comment by jom** [comment]
@GaryJones I'm going to miss my `// WPCS: XSS ok.` comments. Always gave me a chuckle.

**Comment by jom** [comment]
@roundhill @alexsanford I might merge this tomorrow, but if you don't get a chance, feel free to do a post-merge review.

**Pull request merged**